cmake/options.cmake

```diff
@@ -159,6 +159,14 @@ set(ONEDNN_ENABLE_GEMM_KERNELS_ISA "ALL" CACHE STRING
       SSE41 < AVX2 < AVX512 < AMX (or ALL). It means that if user selects, e.g.
       AVX2 ISA, SSE41 kernels will also present at build time.")
 
+set(DNNL_HIP_TARGET_ARCH "" CACHE STRING
+    "Specifies the target offload architecture. Required when compiling
```

Comment:
I kept it vague intentionally as we discussed this will be mainly an internal option, but I have now added a more detailed description (including the link you shared). Let me know if you'd like me to change something.

Reply:
Can you please remind me why we decided to make the option internal?

Reply:
There is one more thing to take care of, the documentation needs to be adjusted. There is a [README.md](https://github.com/intel-innersource/libraries.performance.math.onednn/tree/main/src/gpu/nvidia#readme) file located in `gpu/nvidia` that contains information about the the SYCL kernels and what they cover. Since the SYCL kernels are generic and have a dedicated location we should create a README.md file in `gpu/generic/sycl` and move all the information about the SYCL kernels from the README.md located in `gpu/nvidia` to the new file. And then, we can just add a link to AMD and NVIDIA readme files that would redirect the user to the readme file for the generic SYCL kernels.

Reply:
> Can you please remind me why we decided to make the option internal?

I think @mgouicem suggested it as it will mainly be used for development.

> There is one more thing to take care of, the documentation needs to be adjusted. There is a [README.md](https://github.com/intel-innersource/libraries.performance.math.onednn/tree/main/src/gpu/nvidia?rgh-link-date=2024-08-08T01%3A25%3A30Z#readme) file located in gpu/nvidia that contains information about the the SYCL kernels and what they cover. Since the SYCL kernels are generic and have a dedicated location we should create a README.md file in gpu/generic/sycl and move all the information about the SYCL kernels from the README.md located in gpu/nvidia to the new file. And then, we can just add a link to AMD and NVIDIA readme files that would redirect the user to the readme file for the generic SYCL kernels.

I think I made all the suggested changes. Let me know if I missed something :slightly_smiling_face:

Reply:
> I think @mgouicem suggested it as it will mainly be used for development.

I think I'm starting to remember what the reason was. We consider the option as a temporary solution because it will be removed in the future. Meanwhile we want to test the kernels for AMD vendor so we are going to use the option but I also think the option could be used by the users to enable the kernels and therefore we should not call it an internal option but rather a temporary one. I added a warning to the documentation of the option.

Reply:
> I think I made all the suggested changes. Let me know if I missed something 🙂

The changes look good to me. Thanks for your quick turnaround!